utils/__init__.py

```diff
@@ -264,39 +264,45 @@ def add_xmlns_sbml_attribute(sedml_filepath, sbml_filepath, output_filepath=None
     with open(output_filepath,"w") as fout:
         fout.write(sedstr)
 
-def add_xmlns_fbc_attribute(sedml_filepath, sbml_filepath, output_filepath=None):
+def add_xmlns_fbc_attribute(sedml_filepath, sbml_filepath, temp_sedml_filepath=None):
     '''
-    add an xmlns:sbml attribute to the sedml file that matches the sbml file
-    raise an error if the attribute is already present
-    output fixed file to output_filepath which defaults to sedml_filepath
+    Adds an xmlns:fbc attribute to the SED-ML file. If a temo
+    output fixed file to output_filepath which defaults to sedml_filepath.
+
+    If a temp_sedml_filepath (which could already contain a xmlns:sbml fix) is provided, 
+    this instead of the original SED-ML file is used.
     '''
+    
     sedstr = ""
-    if output_filepath:
-        if os.path.exists(output_filepath):
-            with open(output_filepath, 'r') as file:
+
+    if temp_sedml_filepath:
+        if os.path.exists(temp_sedml_filepath):
+            with open(temp_sedml_filepath, 'r') as file:
                 sedstr = file.read()    
+
     if sedstr == "":
         with open(sedml_filepath, 'r') as file:
             sedstr = file.read()
 
     m = re.search(r'<sedML[^>]*>', sedstr)
-    location = m.span()[1]-1
 
     if m == None:
         raise ValueError(f'Invalid SedML file: main <sedML> tag not found in {sedml_filepath}')
 
-    with open(sbml_filepath, 'r') as file:
-        sbml_str = file.read()
+    else:
```

Comment:
The else isn't really required as the raise will exit the function anyway, might be clearer if you remove it